### PR TITLE
Merge 1.7 fork

### PR DESCRIPTION
**Pull request by KotlinIsland**

### CODE REVIEW

File: mypy/stubgen.py [hunk 1]
```diff
@@ -561,9 +560,14 @@ def _get_func_return(self, o: FuncDef, ctx: FunctionContext) -> str | None:
             if has_return_statement(o):
                 return_name = self.add_name("_typeshed.Incomplete")
             return f"{generator_name}[{yield_name}, {send_name}, {return_name}]"
+        if o.is_property:
+            return None
+
+        if is_trivial_body(o.body) and not self.legacy:
+            return self.add_name("_typeshed.Incomplete")
```
Comment (DetachHead) [inline mypy/stubgen.py:567]: what is this

Comment (KotlinIsland) [reply mypy/stubgen.py:567]:
```py
def foo(): ...
```
has a 'trivial' body, and will generate the stub:
```py
def foo() -> Untyped: ...
```

File: test-data/unit/check-type-aliases.test [hunk 2]
```diff
@@ -256,6 +264,10 @@ def test() -> None:
     x: JSON  # E: Cannot resolve name "JSON" (possible cyclic definition)  \
              # N: Recursive types are not allowed at function scope
     JSON = Union[int, str, List[JSON], Dict[str, JSON]]  # E: Cannot resolve name "JSON" (possible cyclic definition) \
+                                                         # N: Recursive types are not allowed at function scope \
+                                                         # E: Cannot resolve name "JSON" (possible cyclic definition) \
+                                                         # N: Recursive types are not allowed at function scope \
+                                                         # E: Cannot resolve name "JSON" (possible cyclic definition) \
```
[the remark this answers is not in the document]
Comment (KotlinIsland) [reply test-data/unit/check-type-aliases.test:270]: This is an artifact of basedmypy showing multiple errors on the same line that have different columns.

While it's an obvious 'work properly' feature, there exists a couple of cases of upstream code that depend on the 'work not properly' behaviour and report the same error multiple times.

File: mypy/stubgen.py (hunk 1 above)
Comment (KotlinIsland) [reply mypy/stubgen.py:567]: actually, what the spruce was I thinking!

**Review by DetachHead — APPROVED** [review]

**Pull request closed without merging**